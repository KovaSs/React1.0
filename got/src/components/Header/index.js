import React, {Component} from 'react';
import { Button } from 'reactstrap';

import {got} from "../../services/service"
import "./header.css"

class Header extends Component {

	getAllCharacters() {
		return got.getAllCharacters().then(res => console.log(res))
	} 
	getAllHouses() {
		return got.getAllHouses().then(res => console.log(res))
	} 
	getAllBooks() {
		return got.getAllBooks().then(res => console.log(res))
	} 
	render() {
		return (
			<div className="header-block">
				<h3 className="header-title">
					<Button color="link">Game of Thrones DB</Button>
				</h3>
				<ul className="header-links">
					<li>
						<Button 
							color="link"
							onClick={this.getAllCharacters}
						>
							Characters
						</Button>
					</li>
					<li>
						<Button 
							color="link"
							onClick={this.getAllHouses}
						>
							Houses
						</Button>
					</li>
					<li>
						<Button 
							color="link"
							onClick={this.getAllBooks}
						>
							Books
						</Button>
					</li>
				</ul>
			</div>
		);
	}
};

export default Header;