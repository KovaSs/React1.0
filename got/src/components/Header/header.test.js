import React from 'react';
import {shallow} from 'enzyme';
import Header from './index';

describe('Testing <Header/>', () => {
  it('Header have rendered correctly', () => {
    const header = shallow(<Header/>);
    expect(header).toMatchSnapshot();
  })
})
