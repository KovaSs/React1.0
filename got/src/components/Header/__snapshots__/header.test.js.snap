// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <Header/> Header have rendered correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Link
          replace={false}
          to="/"
        >
          <h3
            className="header-title"
          >
            <Button
              color="link"
              tag="button"
            >
              Game of Thrones DB
            </Button>
          </h3>
        </Link>,
        <ul
          className="header-links"
        >
          <li>
            <Button
              color="link"
              onClick={[Function]}
              tag="button"
            >
              Random Character
            </Button>
          </li>
          <Link
            replace={false}
            to="/character/"
          >
            <Button
              color="link"
              onClick={[Function]}
              tag="button"
            >
              Characters
            </Button>
          </Link>
          <Link
            replace={false}
            to="/houses/"
          >
            <Button
              color="link"
              onClick={[Function]}
              tag="button"
            >
              Houses
            </Button>
          </Link>
          <Link
            replace={false}
            to="/books/"
          >
            <Button
              color="link"
              onClick={[Function]}
              tag="button"
            >
              Books
            </Button>
          </Link>
        </ul>,
      ],
      "className": "header-block",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <h3
            className="header-title"
          >
            <Button
              color="link"
              tag="button"
            >
              Game of Thrones DB
            </Button>
          </h3>,
          "replace": false,
          "to": "/",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Button
              color="link"
              tag="button"
            >
              Game of Thrones DB
            </Button>,
            "className": "header-title",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Game of Thrones DB",
              "color": "link",
              "tag": "button",
            },
            "ref": null,
            "rendered": "Game of Thrones DB",
            "type": [Function],
          },
          "type": "h3",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li>
              <Button
                color="link"
                onClick={[Function]}
                tag="button"
              >
                Random Character
              </Button>
            </li>,
            <Link
              replace={false}
              to="/character/"
            >
              <Button
                color="link"
                onClick={[Function]}
                tag="button"
              >
                Characters
              </Button>
            </Link>,
            <Link
              replace={false}
              to="/houses/"
            >
              <Button
                color="link"
                onClick={[Function]}
                tag="button"
              >
                Houses
              </Button>
            </Link>,
            <Link
              replace={false}
              to="/books/"
            >
              <Button
                color="link"
                onClick={[Function]}
                tag="button"
              >
                Books
              </Button>
            </Link>,
          ],
          "className": "header-links",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Button
                color="link"
                onClick={[Function]}
                tag="button"
              >
                Random Character
              </Button>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Random Character",
                "color": "link",
                "onClick": [Function],
                "tag": "button",
              },
              "ref": null,
              "rendered": "Random Character",
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Button
                color="link"
                onClick={[Function]}
                tag="button"
              >
                Characters
              </Button>,
              "replace": false,
              "to": "/character/",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Characters",
                "color": "link",
                "onClick": [Function],
                "tag": "button",
              },
              "ref": null,
              "rendered": "Characters",
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Button
                color="link"
                onClick={[Function]}
                tag="button"
              >
                Houses
              </Button>,
              "replace": false,
              "to": "/houses/",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Houses",
                "color": "link",
                "onClick": [Function],
                "tag": "button",
              },
              "ref": null,
              "rendered": "Houses",
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Button
                color="link"
                onClick={[Function]}
                tag="button"
              >
                Books
              </Button>,
              "replace": false,
              "to": "/books/",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Books",
                "color": "link",
                "onClick": [Function],
                "tag": "button",
              },
              "ref": null,
              "rendered": "Books",
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": "ul",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Link
            replace={false}
            to="/"
          >
            <h3
              className="header-title"
            >
              <Button
                color="link"
                tag="button"
              >
                Game of Thrones DB
              </Button>
            </h3>
          </Link>,
          <ul
            className="header-links"
          >
            <li>
              <Button
                color="link"
                onClick={[Function]}
                tag="button"
              >
                Random Character
              </Button>
            </li>
            <Link
              replace={false}
              to="/character/"
            >
              <Button
                color="link"
                onClick={[Function]}
                tag="button"
              >
                Characters
              </Button>
            </Link>
            <Link
              replace={false}
              to="/houses/"
            >
              <Button
                color="link"
                onClick={[Function]}
                tag="button"
              >
                Houses
              </Button>
            </Link>
            <Link
              replace={false}
              to="/books/"
            >
              <Button
                color="link"
                onClick={[Function]}
                tag="button"
              >
                Books
              </Button>
            </Link>
          </ul>,
        ],
        "className": "header-block",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <h3
              className="header-title"
            >
              <Button
                color="link"
                tag="button"
              >
                Game of Thrones DB
              </Button>
            </h3>,
            "replace": false,
            "to": "/",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Button
                color="link"
                tag="button"
              >
                Game of Thrones DB
              </Button>,
              "className": "header-title",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Game of Thrones DB",
                "color": "link",
                "tag": "button",
              },
              "ref": null,
              "rendered": "Game of Thrones DB",
              "type": [Function],
            },
            "type": "h3",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                <Button
                  color="link"
                  onClick={[Function]}
                  tag="button"
                >
                  Random Character
                </Button>
              </li>,
              <Link
                replace={false}
                to="/character/"
              >
                <Button
                  color="link"
                  onClick={[Function]}
                  tag="button"
                >
                  Characters
                </Button>
              </Link>,
              <Link
                replace={false}
                to="/houses/"
              >
                <Button
                  color="link"
                  onClick={[Function]}
                  tag="button"
                >
                  Houses
                </Button>
              </Link>,
              <Link
                replace={false}
                to="/books/"
              >
                <Button
                  color="link"
                  onClick={[Function]}
                  tag="button"
                >
                  Books
                </Button>
              </Link>,
            ],
            "className": "header-links",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Button
                  color="link"
                  onClick={[Function]}
                  tag="button"
                >
                  Random Character
                </Button>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Random Character",
                  "color": "link",
                  "onClick": [Function],
                  "tag": "button",
                },
                "ref": null,
                "rendered": "Random Character",
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Button
                  color="link"
                  onClick={[Function]}
                  tag="button"
                >
                  Characters
                </Button>,
                "replace": false,
                "to": "/character/",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Characters",
                  "color": "link",
                  "onClick": [Function],
                  "tag": "button",
                },
                "ref": null,
                "rendered": "Characters",
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Button
                  color="link"
                  onClick={[Function]}
                  tag="button"
                >
                  Houses
                </Button>,
                "replace": false,
                "to": "/houses/",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Houses",
                  "color": "link",
                  "onClick": [Function],
                  "tag": "button",
                },
                "ref": null,
                "rendered": "Houses",
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Button
                  color="link"
                  onClick={[Function]}
                  tag="button"
                >
                  Books
                </Button>,
                "replace": false,
                "to": "/books/",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Books",
                  "color": "link",
                  "onClick": [Function],
                  "tag": "button",
                },
                "ref": null,
                "rendered": "Books",
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": "ul",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
